Doc/library/typing.rst

```diff
@@ -36,6 +36,24 @@ New features are frequently added to the ``typing`` module.
 The `typing_extensions <https://pypi.org/project/typing-extensions/>`_ package
 provides backports of these new features to older versions of Python.
 
+Deprecation Timeline of Major Features
+======================================
+
+Certain features in ``typing`` are deprecated and may be removed in a future
+version of Python. The following table summarizes major deprecations for your
+convenience. This is subject to change, and not all deprecations are listed.
+
++----------------------------------+---------------+-------------------+----------------+
+|  Feature                         | Deprecated in | Projected removal | PEP/issue      |
++==================================+===============+===================+================+
+|  ``typing.io`` and ``typing.re`` | 3.8           | 3.12              | :issue:`38291` |
+|  submodules                      |               |                   |                |
++----------------------------------+---------------+-------------------+----------------+
+|  ``typing`` versions of standard | 3.9           | not planned       | :pep:`585`     |
```

Comment:
PEP 585 says:
> The deprecated functionality will be removed from the typing module in the first Python version released 5 years after the release of Python 3.9.0.

Which would probably be 3.14. This timeline always struck me as a little aggressive, given that 3.9 goes EOL in 2025 + libraries won't be able to change code until at least after that happens + this is super breaking.

If we feel differently now, it might be worth updating PEP 585 to avoid confusion.

Reply:
I think I can place a dash `-` there to avoid contradicting the PEP and to avoid committing to a deprecation date that we haven't reached consensus on.